python/eos/analysis.py

```diff
@@ -607,7 +607,7 @@ def _prior_transform(self, u):
         return self._u_to_par(u)
 
 
-    def sample_nested(self, bound='multi', nlive=250, dlogz=1.0, maxiter=None, seed=10, print_progress=True):
+    def sample_nested(self, bound='multi', nlive=250, dlogz=1.0, maxiter=None, seed=10, print_progress=True, save_intermediate=False, base_directory='./', posterior=None):
```

Comment:
That's a reasonable idea to begin with, but I do not like it very much. So far, the design was to separate the tasks from the lower-level operations. I think we should and can keep doing that.

I would suggest to use a ``yield_intermediate`` keyword argument. Instead of using ``return sampler.results`` once at the end, we rather ``yield sampler.results`` periodically. The task can then run a for loop over this function and save the results itself.